clippy_lints/src/use_self.rs

```diff
@@ -49,13 +51,93 @@ impl LintPass for UseSelf {
 
 const SEGMENTS_MSG: &str = "segments should be composed of at least 1 element";
 
+fn span_use_self_lint(cx: &LateContext, path: &Path) {
```

Comment:
This function does the same as 
https://github.com/rust-lang-nursery/rust-clippy/blob/06d6710147742892efc17ed307b3397bfe093bc7/clippy_lints/src/utils/mod.rs#L624-L631